src/app/ui-kit/components/button/button.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { TextName } from '../../text-types';
+
+@Component({
+  selector: 'app-button',
+  templateUrl: './button.component.html',
+})
+export class ButtonComponent implements OnInit {
+  @Input() type: 'accent' | 'ghost' = 'accent';
+  @Input() size: 'sm' | 'md' = 'sm';
+  @Input() iconState: 'next' | 'prev' | 'none' = 'none';
+  @Input() disabled: boolean = false;
+  @Input() textType: TextName = 'P1';
+
+  imageSrc: string = '../../..//assets/images/chevron.svg';
+  iconDeg: string = '';
+  iconStyles: { [klass: string]: number | string } = {};
+
+  get classes(): string {
+    return [
+      this.iconState === 'prev'
+        ? 'flex-row-reverse pl-3'
```

Comment:
It's hard to read, let's divide this into multiple variables and combine this in an array. And if the ternary operator has multiple layers also divide them